crypto/evp/p_lib.c

```diff
@@ -986,6 +986,35 @@ int evp_pkey_name2type(const char *name)
 
     return type;
 }
+
+/*
+ * Note this method is used by the decoder so the names returned must match
+ * the decoder names.
+ */
+const char *evp_pkey_type2name(int type)
+{
+    switch (type) {
+    case EVP_PKEY_RSA:
+        return "RSA";
+    case EVP_PKEY_RSA_PSS:
+        return "RSA-PSS";
+    case EVP_PKEY_DH:
+        return "DH";
+    case EVP_PKEY_DHX:
+        return "DHX";
```

Comment:
You can use "X9.42 DH" as well...  the decoder method use the same namemap as EVP, so all names should apply.  If you want to be super safe, decoders.inc could use the name string "DHX:X9.42 DH" instead of just "DHX".